dashboard/test/lib/services/user/password_resetter_by_username_test.rb

```diff
@@ -0,0 +1,30 @@
+require 'test_helper'
+
+class Services::User::PasswordResetterByUsernameTest < ActiveSupport::TestCase
+  let(:described_instance) {described_class.new(username: username)}
+
+  let(:username) {Faker::Internet.unique.username(specifier: 5..19)}
+
+  describe '#call' do
+    subject(:reset_password) {described_instance.call}
+
+    let(:mail) {ActionMailer::Base.deliveries.first}
+
+    context 'for username with an existing user' do
+      let!(:user) {create(:user)}
+      before do
+        user.username = username
+        user.save!
+      end
```

Comment:
https://github.com/code-dot-org/code-dot-org/blob/4dc74fea6d0761236a94266b54bf75afec0f1641/dashboard/app/models/user.rb#L573
https://github.com/code-dot-org/code-dot-org/blob/4dc74fea6d0761236a94266b54bf75afec0f1641/dashboard/app/models/user.rb#L1652-L1656

Reply:
Exactly, it is overwriting the username you pass in. A cleaner approach would be to avoid forcing a rewrite of the `username` attribute in the first place, but changing this behavior is not part of this task:
```ruby
self.username ||= UserHelpers.generate_username(User.with_deleted, name)
```